controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -773,3 +777,27 @@ func (r *KubeadmControlPlaneReconciler) adoptOwnedSecrets(ctx context.Context, k
 
 	return nil
 }
+
+// ensureCertificatesOwnerRef ensures an ownerReference to the owner is added on the Secrets holding certificates.
+func ensureCertificatesOwnerRef(ctx context.Context, ctrlclient client.Client, clusterKey client.ObjectKey, certificates secret.Certificates, owner metav1.OwnerReference) error {
+	for _, c := range certificates {
+		s := &corev1.Secret{}
+		secretKey := client.ObjectKey{Namespace: clusterKey.Namespace, Name: secret.Name(clusterKey.Name, c.Purpose)}
+		if err := ctrlclient.Get(ctx, secretKey, s); err != nil {
+			// If the secret isn't found ignore the error.
+			if apierrors.IsNotFound(err) {
+				continue
+			}
+			return errors.Wrapf(errors.WithStack(err), "failed to get Secret %s", secretKey)
```

Comment:
Looks like errors.Wrapf already does errors.WithStack internally. Is there a reason why we have to do it in this case additionally? (I'm not exactly sure how the result is different if we additionally call WithStack here)


(same below)